date.go

```diff
@@ -241,3 +241,53 @@ func Min(d1, d2 Date) Date {
 	}
 	return d2
 }
+
+// NullDate represents a nullable date with compatibility for
+// scanning null values from the database.
+type NullDate struct {
```

Comment:
I've found it helpful when writing third party package code to verify interface compliance. Something like;

```golang
type NullDateSqlField interface {
	sql.Scanner
	driver.Valuer
}

var _ NullDateSqlField = NullDate{}
```

Reply:
updated, thanks, that was really useful